doc/read-the-docs-site/architecture/runtime-view.rst

```diff
@@ -0,0 +1,156 @@
+Runtime view
+============
+
+There are two runtime perspectives in a Marconi application:
+
+* user querying
+* indexing
+
+They will typically run on separate threads (or separate processes) in a Marconi application.
+In the followings sections, we will describe the runtime behavior of each of those perspectives.
+
+Indexing view
+-------------
+
+In this section, we'll show the indexing runtime view of a Marconi application.
+Note that the indexing is mainly handled by the `marconi-core` and `marconi-cardano-core` packages.
+
+.. uml::
+
+    !include <C4/C4_Dynamic>
+    title Marconi indexing runtime view
+
+    Container_Ext(datasource, "Datasource", "", "Represents some external data source that Marconi will connect to")
+
+    Container_Boundary(marconiContainer, "Marconi application") {
+      Component(sourceReader, "Input source reader", "Haskell component", "Module that processes data from the external datasource and transforms it to a [ProcessedInput]")
+      Component(preprocessor, "Preprocessor", "Haskell component", "Applies some preprocessing function to the [ProcessedInput]")
+      Component(transformer, "Transformer", "Haskell component", "Applies some function to the incoming ProcessedInput and forwards them to the indexer")
+      Component(indexer, "Indexer", "Haskell thread", "Indexes the relevant parts of the ProcessedInput to some storage")
+      ContainerDb(marconiDb, "Marconi storage", "Storage backend of the indexer")
+    }
+
+    Rel(datasource, sourceReader, "sends some input data")
+    Rel(sourceReader, preprocessor, "sends '[ProcessedInput]'")
+    Rel(preprocessor, transformer, "sends preprocessed '[ProcessedInput]'")
+    Rel(transformer, indexer, "forwards each 'ProcessedInput'")
+    Rel(indexer, marconiDb, "stores events in the indexer storage")
+
+The input source reader provides an interface to fetch data from any datasource: a local Cardano node, Ogmios, cardano-db-sync, etc.
+However, the input data needs to be mapped to a ``ProcessedInput``.
+The ``ProcessedInput`` is sent to the preprocessor which applies some preprocessing function on the inputs.
+The preprocessed inputs are sent to the transformer which also transforms the inputs, and sends them to the indexer.
+Finally, the indexer will index them to it's storage backend.
+
+.. note::
+
+  See the explanation of the `marconi-core` components in order to understand concepts like ``Preprocessor``,  ``Transformer``, ``Indexer``, etc..
```

Comment:
I realize that this isn't merged yet, would be good to provide a link after it is